src/app/layout/contact-new/contact-new.component.html

```diff
@@ -0,0 +1,55 @@
+<!-- contact-new.component.html -->
+<h2>Add New Contact</h2>
+<form (ngSubmit)="onSubmit()" #contactForm="ngForm">
+  <div>
+    <label for="firstName">First Name</label>
+    <input
+      type="text"
+      id="firstName"
+      name="firstName"
+      [(ngModel)]="firstName"
+      required
+      minlength="3"
+      pattern="^[^\s]+$"
+    >
+    <div *ngIf="contactForm.controls['firstName'].invalid && (contactForm.controls['firstName'].dirty || contactForm.controls['firstName'].touched)">
+      <div class="error-message">First name must be at least 3 characters long and should not contain spaces.</div>
+    </div>
+  </div>
+  <div>
+    <label for="lastName">Last Name</label>
+    <input
+      type="text"
+      id="lastName"
+      name="lastName"
+      [(ngModel)]="lastName"
+      required
```

Comment:
Just be aware that this is Template driven forms, and is different to ReactiveForms. There are pros and cons to both.

Reply:
Appreciate the feedback, I'll be sure to take note.